File: toeic-mock-test-web/src/features/user/types/user-response.ts
 import { RecordStatusType } from "@/types/common.type";

    export interface UserResponse {
    id: string;
    name: string;
    email: string;
    firstName: string;
    lastName: string;
    fullName: string; 
    roleId: string;
    roleName: string;
    status: RecordStatusType;
    createdDate: string; 
    }

    export interface UserDetailResponse extends UserResponse {
    createdById?: string;
    updatedDate?: string;
    updatedById?: string;
    }

File: toeic-mock-test-api/ToeicMockTest.API/ToeicMockTest.Contracts/Users/Responses/UserResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToeicMockTest.SharedKernel.Common.Enums;

namespace ToeicMockTest.Contracts.Users.Responses
{
    public class UserResponse
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string FullName => $"{FirstName} {LastName}";
        public Guid RoleId { get; set; }
        public string RoleName { get; set; } = string.Empty; 
        public RecordStatus Status { get; set; }
        public DateTime CreatedDate { get; set; }
    }
}
